// @crossmarkio/utils/dist/src/modules/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Modules = void 0;
const typings_1 = __importDefault(require("@crossmarkio/typings"));
const Config = typings_1.default.crossmark.config.Config.config;
class Modules {
    opts;
    isProd;
    browser;
    constructor(opts) {
        this.opts = opts;
        this.isProd = this.opts.run === 'production';
        this.browser = this.isProd ? (chrome ? chrome : require('webextension-polyfill')) : '';
    }
    title = (t) => {
        let title = t || '';
        let version = this.extension.getVersion();
        if (this.opts.dev)
            title = title.concat(' ', '- DEV');
        if (this.opts.beta)
            title = title.concat(' ', '- BETA');
        if (version && (this.opts.beta || this.opts.dev))
            title = title.concat(' ', `v${version}`);
        return title;
    };
    extension = {
        getExtensionId: () => {
            if (this.isProd)
                return chrome.runtime.id;
            return 'crossmark';
        },
        getVersion: () => {
            if (this.isProd)
                return this.browser.runtime.getManifest().version;
            return this.opts.version || `0.0.0`;
        },
    };
}
exports.Modules = Modules;
