// @crossmarkio/utils/dist/src/helpers/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.abbreviate = exports.string = exports.sleep = exports.uuid = exports.site = exports.crypto = exports.hexConversion = void 0;
const hexConversion = __importStar(require("./hexConversion"));
exports.hexConversion = hexConversion;
const crypto = __importStar(require("../lib/crypto"));
exports.crypto = crypto;
const site = __importStar(require("./siteData"));
exports.site = site;
const string = __importStar(require("./string"));
exports.string = string;
const abbreviate = __importStar(require("./abbreviate"));
exports.abbreviate = abbreviate;
const identifiers_1 = require("./identifiers");
Object.defineProperty(exports, "uuid", { enumerable: true, get: function () { return identifiers_1.uuid; } });
const sleep_1 = require("./sleep");
Object.defineProperty(exports, "sleep", { enumerable: true, get: function () { return sleep_1.sleep; } });
