// @crossmarkio/utils/dist/src/lib/xrpl/rpc/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ledger_entry = exports.tx = exports.account_offers = exports.account_nfts = exports.account_tx = exports.account_objects = exports.account_currencies = exports.server_info = exports.account_info = exports.account_lines = void 0;
const axios_1 = __importDefault(require("axios"));
const methods_1 = __importDefault(require("./methods"));
const account_lines = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_lines(params)))
    .data;
exports.account_lines = account_lines;
const account_info = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_info(params))).data;
exports.account_info = account_info;
const server_info = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.server_info())).data;
exports.server_info = server_info;
const account_currencies = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_currencies(params)))
    .data;
exports.account_currencies = account_currencies;
const account_objects = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_objects(params)))
    .data;
exports.account_objects = account_objects;
const account_tx = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_tx(params))).data;
exports.account_tx = account_tx;
/* export const ledger = ({
  params.rpc,
  input,
}: {
  params.rpc: string;
  input: LedgerRequestInput;
}) => axios.post(params.rpc, methods.ledger(input)); */
const account_nfts = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_nfts(params))).data;
exports.account_nfts = account_nfts;
const account_offers = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.account_nfts(params)))
    .data;
exports.account_offers = account_offers;
const tx = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.tx(params))).data;
exports.tx = tx;
const ledger_entry = async (params) => (await axios_1.default.post(params.rpc, methods_1.default.ledger_entry(params))).data;
exports.ledger_entry = ledger_entry;
exports.default = {
    server_info: exports.server_info,
    account_info: exports.account_info,
    account_lines: exports.account_lines,
    account_objects: exports.account_objects,
    account_tx: exports.account_tx,
    account_nfts: exports.account_nfts,
    ledger_entry: exports.ledger_entry,
    account_offers: exports.account_offers,
    tx: exports.tx,
};
