// @crossmarkio/typings/dist/src/schema/v2/api/defaults/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultNodes = exports.nodes = exports.uuid = void 0;
const protocols_1 = require("../../protocols");
const networks_1 = require("../../networks");
const uuid = () => {
    let uuid = '', ii;
    for (ii = 0; ii < 32; ii += 1) {
        switch (ii) {
            case 8:
            case 20:
                uuid += '-';
                uuid += ((Math.random() * 16) | 0).toString(16);
                break;
            case 12:
                uuid += '-';
                uuid += '4';
                break;
            case 16:
                uuid += '-';
                uuid += ((Math.random() * 4) | 8).toString(16);
                break;
            default:
                uuid += ((Math.random() * 16) | 0).toString(16);
        }
    }
    return uuid;
};
exports.uuid = uuid;
exports.nodes = [
    {
        id: 0,
        uuid: (0, exports.uuid)(),
        name: 'xrplf',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.mainnet,
        status: undefined,
        wss: 'wss://xrplcluster.com/',
        rpc: 'https://xrplcluster.com/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 0,
        uuid: (0, exports.uuid)(),
        name: 'ripple s1',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.mainnet,
        status: undefined,
        wss: 'wss://s1.ripple.com/',
        rpc: 'https://s1.ripple.com:51234/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 0,
        uuid: (0, exports.uuid)(),
        name: 'ripple s2',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.mainnet,
        status: undefined,
        wss: 'wss://s2.ripple.com/',
        rpc: 'https://s2.ripple.com:51234/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 1,
        uuid: (0, exports.uuid)(),
        name: 'ripple',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.testnet,
        status: undefined,
        wss: 'wss://s.altnet.rippletest.net:51233/',
        rpc: 'https://s.altnet.rippletest.net:51234/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 1,
        uuid: (0, exports.uuid)(),
        name: 'xrpl-labs',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.testnet,
        status: undefined,
        wss: 'wss://testnet.xrpl-labs.com/',
        rpc: 'https://testnet.xrpl-labs.com/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 2,
        uuid: (0, exports.uuid)(),
        name: 'ripple',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.devnet,
        status: undefined,
        wss: 'wss://s.devnet.rippletest.net:51233/',
        rpc: 'https://s.devnet.rippletest.net:51234/',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 0,
        uuid: (0, exports.uuid)(),
        name: 'xmark',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.mainnet,
        status: undefined,
        wss: 'wss://m.cluster.crossmark.dev',
        rpc: 'https://m.cluster.crossmark.dev/rpc',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: 1,
        uuid: (0, exports.uuid)(),
        name: 'xmark',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.testnet,
        status: undefined,
        wss: 'wss://t.cluster.crossmark.dev',
        rpc: 'https://t.cluster.crossmark.dev/rpc',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: null,
    },
    {
        id: null,
        uuid: (0, exports.uuid)(),
        name: 'amm',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xrpl,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.xls30d,
        status: undefined,
        wss: 'wss://amm.devnet.rippletest.net:51233',
        rpc: 'http://amm.devnet.rippletest.net:51234',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xrplExpl.bithomp,
        faucet: 'https://faucet.amm.rippletest.net/accounts',
        docs: 'https://dev.to/ripplexdev/xrpl-amm-network-available-for-testing-and-development-426i',
    },
    {
        id: null,
        uuid: (0, exports.uuid)(),
        name: 'xrplf',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xahau,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.mainnet,
        status: undefined,
        wss: 'wss://xahau.network',
        rpc: 'https://xahau.network',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xahauExpl.xrplf,
        faucet: undefined,
        docs: 'https://xrpl-hooks.readme.io/',
    },
    {
        id: null,
        uuid: (0, exports.uuid)(),
        name: 'xrpl-labs',
        protocol: protocols_1.Protocols.xrpl,
        network: networks_1.AllNetworks.NetworkLabels.xahau,
        type: networks_1.AllNetworks.AllNodeTypes.xrpl.testnet,
        status: undefined,
        wss: 'wss://hooks-testnet-v3.xrpl-labs.com',
        rpc: 'https://hooks-testnet-v3.xrpl-labs.com',
        explorer: networks_1.AllNetworks.AllNodeExplorers.xahauExpl.xrplf,
        faucet: undefined,
        docs: 'https://xrpl-hooks.readme.io/',
    },
];
exports.defaultNodes = exports.nodes;
