// @crossmarkio/typings/dist/src/schema/v1/api/cards.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardSchema = exports.CardColor = exports.TWCardColor = void 0;
var TWCardColor;
(function (TWCardColor) {
    TWCardColor["color1"] = "tw-bg-cardGradient1";
    TWCardColor["color2"] = "tw-bg-cardGradient2";
    TWCardColor["color3"] = "tw-bg-cardGradient3";
    TWCardColor["color4"] = "tw-bg-cardGradient4";
    TWCardColor["color5"] = "tw-bg-cardGradient5";
    TWCardColor["color6"] = "tw-bg-cardGradient6";
    TWCardColor["color7"] = "tw-bg-cardGradient7";
    TWCardColor["color8"] = "tw-bg-cardGradient8";
    TWCardColor["color9"] = "tw-bg-cardGradient9";
    TWCardColor["color10"] = "tw-bg-cardGradient10";
})(TWCardColor || (exports.TWCardColor = TWCardColor = {}));
var CardColor;
(function (CardColor) {
    CardColor["color1"] = "color1";
    CardColor["color2"] = "color2";
    CardColor["color3"] = "color3";
    CardColor["color4"] = "color4";
    CardColor["color5"] = "color5";
    CardColor["color6"] = "color6";
    CardColor["color7"] = "color7";
    CardColor["color8"] = "color8";
    CardColor["color9"] = "color9";
    CardColor["color10"] = "color10";
})(CardColor || (exports.CardColor = CardColor = {}));
class CardSchema {
    id = '';
    address;
    label = 'Personal Account';
    source;
    addresses = [];
    themes = [];
    accessLevel;
    securityLevel;
    encryption;
    default = false;
    registeredAt = Date.now();
    createdAt = Date.now();
    updatedAt = Date.now();
    static migration = (oldValue, newValue) => { };
}
exports.CardSchema = CardSchema;
