// @crossmarkio/typings/dist/src/schema/v1/api/security.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecuritySchema = void 0;
class SecuritySchema {
    passcode;
    password;
    recovery;
    autolock = false;
    firewall = false;
    locked = false;
    minutesAutoLock;
    lastPasscodeFailedTimestamp = Date.now();
    passcodeFailedAttempts;
    lastUnlockedTimestamp = Date.now();
    purgeOnBruteForce = false;
    attemptsBeforePurge = 5;
    passcodeFallback;
    derivation;
    static schema = {
        passcode: 'string?',
        password: 'string?',
        recovery: 'string?',
        updatedAt: 'Date',
        minutesAutoLock: { type: 'int', default: 1 },
        lastPasscodeFailedTimestamp: 'int?',
        passcodeFailedAttempts: { type: 'int', default: 0 },
        lastUnlockedTimestamp: 'int?',
        purgeOnBruteForce: { type: 'bool', default: false },
        passcodeFallback: { type: 'bool', default: false }, // fallback to passcode in case of biometric fail
    };
    static migration = (oldValue, newValue) => { };
}
exports.SecuritySchema = SecuritySchema;
